src/renderer/entities/wallet/ui/CheckPermission/CheckPermissionWithRedirect.tsx

```diff
@@ -0,0 +1,33 @@
+import { PropsWithChildren } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { Account, Wallet } from '@shared/core';
+import { OperationType } from '../../common/types';
+import { getOperationTypeFn } from '../../common/utils';
+
+type Props = {
+  operationType: OperationType;
+  wallet?: Wallet;
+  accounts: Account[];
+  redirectPath: string;
+};
+
+export const CheckPermissionWithRedirect = ({
+  operationType,
+  wallet,
+  accounts,
+  children,
+  redirectPath,
+}: PropsWithChildren<Props>): JSX.Element => {
+  const navigate = useNavigate();
+
+  const operationFn = getOperationTypeFn(operationType);
+
+  if (wallet && operationFn(wallet, accounts)) {
+    return <>{children}</>;
+  } else {
+    navigate(redirectPath);
```

Comment:
As I remember react-router provides us `<Redirect />` component to be rendered when you require redirect
`navigate` is not supposed to be used instead of rendering component